arviz/data/io_pymc3.py

```diff
@@ -125,27 +129,38 @@ def posterior_predictive_to_xarray(self):
     @requires("prior")
     def prior_to_xarray(self):
         """Convert prior samples to xarray."""
-        return dict_to_dataset(
-            {k: np.expand_dims(v, 0) for k, v in self.prior.items()},
-            library=self.pymc3,
-            coords=self.coords,
-            dims=self.dims,
-        )
+        # workaround due to pymc#3588, to do, divide between prior and prior predictive
+        # probably after solving this issue or while solving it.
+        model_vars = []
+        if self.trace is not None:
```

Comment:
it looks like you construct `model_vars` here and in `constant_data_to_xarray`, but protect against different things in each (for example, I am nervous that we check whether `self.trace is not None` here, but not there).  It is also a little hard to figure out that `constant_data_to_xarray` wants all the elements *not* in `model_vars`.

Maybe factor out a small helper function to construct that list?